pkg/operation/botanist/controlplane/kubecontrollermanager/kube_controller_manager.go

```diff
@@ -0,0 +1,529 @@
+// limitations under the License.
+
+package kubecontrollermanager
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/secrets"
+
+	"github.com/Masterminds/semver"
+	resourcesv1alpha1 "github.com/gardener/gardener-resource-manager/pkg/apis/resources/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	autoscalingv1beta2 "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1beta2"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	// ServiceName is the name of the service of the kube-controller-manager.
+	ServiceName = "kube-controller-manager"
+
+	// SecretNameKubeconfig is a constant for the secret name for the kube-controller-manager's kubeconfig secret.
+	SecretNameKubeconfig = "kube-controller-manager"
+	// SecretNameServer is the name of the kube-controller-managerr server certificate secret.
+	SecretNameServer = "kube-controller-manager-server"
+
+	// labelRole is a constant for the value of a label with key 'role' whose value is 'controller-manager'.
+	labelRole = "controller-manager"
+	// portNameMetrics is a constant for the name of the metrics port of the kube-controller-manager.
+	portNameMetrics = "metrics"
+
+	// volumeMountPathCA is the volume mount path for the CA certificate used by the kube controller manager.
+	volumeMountPathCA = "/srv/kubernetes/ca"
+	// volumeMountPathCA is the volume mount path for the service account key that is a PEM-encoded private RSA or ECDSA key used to sign service account tokens.
+	volumeMountPathServiceAccountKey = "/srv/kubernetes/service-account-key"
+	// volumeMountPathCA is the volume mount path for the kubeconfig which can be used by the kube-controller-manager to communicate with the kube-apiserver.
+	volumeMountPathKubeconfig = "/var/lib/kube-controller-manager"
+	// volumeMountPathCA is the volume mount path for the x509 TLS server certificate and key for the HTTPS server inside the kube-controller-manager (which is used for metrics and health checks).
+	volumeMountPathServer = "/var/lib/kube-controller-manager-server"
+
+	// ShootManagedResourceName is the name of the managed resource that contains the resources to be deployed into the Shoot cluster.
+	ShootManagedResourceName = "shoot-core-kube-controller-manager"
```

Comment:
Why is this exported?

Reply:
Used in the kube controller manager test

Reply:
There is no need to expose a constant just for a unit test. Please make it unexported.